server/openapi_server/controllers/challenge_controller.py

```diff
@@ -44,30 +43,30 @@ def create_challenge(challenge=None):
                 except DoesNotExist:
                     status = 404
                     res = Error(f"Person {person_id} not found", status)
+                    return res, status
 
             # Add challenge if all tags and persons are found and valid.
-            if status is None:
-                try:
-                    db_challenge = DbChallenge(
-                        name=challenge.name,
-                        startDate=challenge.start_date,
-                        endDate=challenge.end_date,
-                        url=challenge.url,
-                        status=challenge.status,
-                        # grant=challenge.grant,
-                        organizers=challenge.organizers,
-                        tags=challenge.tags,
-                        challengeResults=DbChallengeResults(
-                            nSubmissions=challenge.challenge_results.n_submissions,  # noqa: E501
-                            nFinalSubmissions=challenge.challenge_results.n_final_submissions,  # noqa: E501
-                            nRegisteredParticipants=challenge.challenge_results.n_registered_participants,  # noqa: E501
-                        )
-                    ).save(force_insert=True)
-                    res = Challenge.from_dict(db_challenge.to_dict())
-                    status = 200
-                except NotUniqueError as error:
-                    status = 409
-                    res = Error("Conflict", status, str(error))
+            try:
+                db_challenge = DbChallenge(
+                    name=challenge.name,
+                    startDate=challenge.start_date,
+                    endDate=challenge.end_date,
+                    url=challenge.url,
+                    status=challenge.status,
+                    # grant=challenge.grant,
+                    organizers=challenge.organizers,
+                    tags=challenge.tags,
+                    challengeResults=DbChallengeResults(
+                        nSubmissions=challenge.challenge_results.n_submissions,  # noqa: E501
+                        nFinalSubmissions=challenge.challenge_results.n_final_submissions,  # noqa: E501
+                        nRegisteredParticipants=challenge.challenge_results.n_registered_participants,  # noqa: E501
+                    )
+                ).save(force_insert=True)
+                res = ChallengeCreateResponse.from_dict(db_challenge.to_dict())
```

Comment:
Does this return a payload that has only the property `challengeId`? If yes, this line makes an assumption that the properties of `ChallengeCreateResponse` (there's actually only one property) are a subset of the properties of `Challenge`. This assumption may lead to unforeseen effect in the future. The code would be more robust if we create the object `ChallengeCreateResponse` using its constructor.

Reply:
@tschaffter yes, the returned payload contains just the `challengeId`:

<img width="424" alt="Screen Shot 2021-01-21 at 5 28 18 PM" src="https://user-images.githubusercontent.com/9377970/105433224-136adc80-5c0e-11eb-8960-d958943eb1c0.png">

> This assumption may lead to unforeseen effect in the future. The code would be more robust if we create the object `ChallengeCreateResponse` using its constructor.

Thanks for the insight! I didn't think of that.